### ISSUE TITLE
Configurable merge order in addConfig()

### ISSUE THREAD

**Issue by kevinkace** [tivac/dullard#108]
CLI args and `includes` config options are both merged with `addConfig()`, but the source and destination is not taken into account wrt the base `.dullfile`.

https://github.com/tivac/dullard/blob/master/src/dullard.js#L291
https://github.com/tivac/dullard/blob/master/src/cli.js#L64
https://github.com/tivac/dullard/blob/master/src/dullard.js#L236

Eg:

```js
// ./.dullfile
{
  "program" : "local",
  "includes" : [ "./inc/.dullfile" ]
}
// ./inc/.dullfile
{
  "program" : "live"
}
```
Run from CLI `dullard --program=dev`

Results:
`v1.1.0` - `program === dev`
`v1.1.1` - `program === local`

**Comment by tivac** [comment]
This seems really bad, I'll need to fix asap and deprecate `1.1.1`

**Comment by kevinkace** [comment]
Here's a solution with adding a `isSource` to `addConfig()`

https://github.com/tivac/dullard/compare/master...kevinkace:master
